src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormHistoricalBalance.js

```diff
@@ -12,26 +12,48 @@ import {
 import { ASSETS_BY_WALLET_QUERY } from '../../../../HistoricalBalance/common/queries'
 import {
   mapAssetsHeldByAddressToProps,
-  mapToAssets,
   isPossibleEthAddress
 } from '../../../utils/utils'
+import { TriggerProjectsSelector } from './ProjectsSelector/TriggerProjectsSelector'
+import FormikSelect from '../../../../../components/formik-santiment-ui/FormikSelect'
 import styles from '../signal/TriggerForm.module.scss'
 
-const isDisabledWalletAddressField = (
-  canUseMappedErc20,
-  target,
-  allErc20Projects
-) => {
-  if (canUseMappedErc20 || !target || !target.value) {
-    return false
+const isInHeldAssets = (heldAssets, checking) => {
+  for (let i = 0; i < checking.length; i++) {
+    const checkingAsset = checking[i]
+
+    if (!heldAssets.some(({ slug }) => slug === checkingAsset.slug)) {
+      return false
+    }
   }
+  return true
+}
 
-  return !(
+const isNotErc20Assets = (target, allErc20Projects) => {
+  const isEthOrErc20 =
     target.value === 'ethereum' ||
-    allErc20Projects.some(x => x.value === target.value)
-  )
+    (Array.isArray(target) && isInHeldAssets(allErc20Projects, target)) ||
+    allErc20Projects.some(({ slug }) => slug === target.slug)
+
+  return !isEthOrErc20
 }
 
+const mapAssetsToAllProjects = (all, heldAssets) => {
+  let items = []
+  for (let i = 0; i < heldAssets.length; i++) {
+    const item = heldAssets[i]
```

Comment:
the same about `forEach`

Reply:
@GarageInc Rewrite it using reduce. In the prev example using for or for each is ok, cause you can optimize your cycle, but here reduce will be shorter and better for js developer readability.

Reply:
Correct, will fix